src/CustomIndexing/CustomIndexing.jl

```diff
@@ -0,0 +1,130 @@
+module CustomIndexing
+
+"""
+    struct SparseArray{T,N} <: AbstractArray{T,N}
+        data::Dict{Tuple,T}
```

Comment:
Tuple` could be `NTuple{N}`.

Reply:
Not if indices have different types, e.g. `(1, 1.0) isa NTuple{2}` -> `false`

Reply:
In fact we can use `NTuple{N, Any}` but not `NTuple{N}`, see https://discourse.julialang.org/t/calling-a-funcion-with-tuple-types/1745/2
```julia
julia> Tuple{Int, Symbol} <: NTuple{2}
false

julia> Tuple{Int, Symbol} <: NTuple{2, Any}
true
```